### ISSUE TITLE
Switch to 'channels last' once and for all

### ISSUE THREAD

**Issue by mingyip** [rpautrat/SuperPoint#134]
https://github.com/rpautrat/SuperPoint/blob/2cb02a00d952e2b28d8b05ef89005a66ab1f0e61/superpoint/models/super_point.py#L65

# Switch to 'channels last' once and for all
        if config['data_format'] == 'channels_first':
            logits = tf.transpose(logits, [0, 2, 3, 1])

Shouldn't it be 
if config['data_format'] == 'channels_last': ?

**Comment by rpautrat** [comment]
Hi,

The input image is always considered as 'channel_last', and the flag 'channel_first' only means that the forward pass of the network will be done with NCHW instead of the default NHWC. So if config['data_format'] == 'channels_first', the output of the network will be in NCHW format.

Now the computation of the loss is only in channel last format, so if config['data_format'] == 'channels_first' you need to put the channels at the back before computing the loss.

This feature is not very useful anyway, it might just be of some use if one of the two formats is only available for some operations, or when one format is faster than the other.
